apps/server/src/modules/task/service/task.service.spec.ts

```diff
@@ -79,13 +79,38 @@ describe('TaskService', () => {
 	});
 
 	describe('findBySingleParent', () => {
+		const courseId = 'courseId';
+		const creatorId = 'user-id';
+		beforeEach(() => {
+			user = userFactory.buildWithId();
+			authorizationService.getUserWithPermissions.mockResolvedValue(user);
+		});
+		afterEach(() => {
+			authorizationService.hasAllPermissions.mockRestore();
+			authorizationService.getUserWithPermissions.mockRestore();
+		});
 		it('should call findBySingleParent from task repo', async () => {
```

Comment:
Test structure must be updated. 
Remove beforeEach and the const courseId, creatorId. 
Add describes with when .... add to each when a setup that include data generation and mock.